test/metabase/api/notify_test.clj

```diff
@@ -1,28 +1,83 @@
 (ns metabase.api.notify-test
-  (:require [clj-http.client :as client]
+  (:require [cheshire.core :as json]
+            [clj-http.client :as client]
             [clojure.test :refer :all]
+            [environ.core :as env]
             [metabase.http-client :as http]
+            [metabase.models.database :as database]
             [metabase.server.middleware.util :as middleware.u]
-            [metabase.test.fixtures :as fixtures]))
+            [metabase.test :as mt]
+            [metabase.test.fixtures :as fixtures]
+            [metabase.util :as u]))
 
 (use-fixtures :once (fixtures/initialize :db :web-server))
 
+(defn- do-with-api-key [api-key thunk]
+  (with-redefs [env/env (assoc env/env :mb-api-key api-key)]
+    (thunk)))
+
+(defmacro ^:private with-api-key [api-key & body]
+  `(do-with-api-key ~api-key (fn [] ~@body)))
```

Comment:
Mentioned in Slack yesterday, but you should use `mt/with-temporary-settings-values` instead